client/state/data-layer/wpcom/sites/comments-tree/index.js

```diff
@@ -25,24 +25,41 @@ export const fetchCommentsTreeForSite = ( { dispatch }, action ) => {
 			{
 				method: 'GET',
 				path: `/sites/${ siteId }/comments-tree`,
-				apiVersion: '1',
+				apiVersion: '1.1',
 				query: {
-					status: 'unapproved' === status ? 'pending' : status,
+					status,
 				},
 			},
 			action
 		)
 	);
 };
 
-const mapTree = ( tree, status, type ) =>
-	map( tree, ( [ commentId, postId, commentParentId ] ) => ( {
+const mapPosts = ( commentIds, postId ) => {
+	postId = parseInt( postId, 10 );
+
+	const topLevelComments = map( commentIds[ 0 ], commentId => [ commentId, postId, 0 ] );
+
+	const commentReplies = commentIds[ 1 ]
+		? map( commentIds[ 1 ], ( [ commentId, commentParentId ] ) => [
+				commentId,
+				postId,
+				commentParentId,
+			] )
+		: [];
```

Comment:
Note that `map()` is safe for empty lists. We could simplify if we just do `( commentIds[ 1 ] || [] ).map( … )`

Reply:
What do you think if we simplified this ternary with an early return instead?

```es6
if ( ! commentIds[ 1 ] ) {
  return [];
}

return map( commentIds[ 1 ], ... );
```

It would be equally readable if we destructured `commentIds` into named variables.

(To clarify, I'm mostly proposing this because currently Prettier has a hard time with this kind of ternaries indentation, and the editor errors bug me a lot 😄 )

Reply:
Put into pictures…

```js
const [ topLevel, replies ] = commentIds;

return [
	topLevel.map( id => [ id, postId, 0 ] ),
	replies.map( ( [ id, parentId ] ) => [ id, postId, parentId ] ),
]
```

Reply:
What a show off! 😛